packages/samples/headless-react/src/components/search-parameter-manager/search-parameter-manager.ts

```diff
@@ -0,0 +1,30 @@
+import {
+  buildSearchParameterManager,
+  buildSearchParameterSerializer,
+  Engine,
+} from '@coveo/headless';
+
+const {serialize, deserialize} = buildSearchParameterSerializer();
+
+function readSearchParametersFromURI() {
+  return deserialize(decodeURIComponent(window.location.hash.slice(1)));
+}
+
+type SearchParameters = ReturnType<typeof readSearchParametersFromURI>;
+
+function writeSearchParametersToURI(searchParameters: SearchParameters) {
+  window.location.hash = serialize(searchParameters);
+}
+
+export function bindSearchParametersToURI(engine: Engine) {
+  const searchParameterManager = buildSearchParameterManager(engine, {
+    initialState: {parameters: readSearchParametersFromURI()},
+  });
+
+  return {
+    autoUpdateURI: () =>
+      searchParameterManager.subscribe(() =>
+        writeSearchParametersToURI(searchParameterManager.state.parameters)
+      ),
+  };
+}
```

Comment:
Unlike production code, I feel sample code is much easier to read if there was only one function. I would suggest:

```
export function bindSearchParametersToURI(engine: Engine) {
    const hash = window.location.hash.slice(1);
    const parameters = deserialize(decodeURIComponent(hash))

    const manager = buildSearchParameterManager(engine, {initialState: {parameters}};

   ...
}
```